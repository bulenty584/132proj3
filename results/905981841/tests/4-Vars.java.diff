1,8c1,3
< Exception in thread "main" java.lang.RuntimeException: Error: v7 = call v6(v1)
< Wrong argument type in Call
< 	at sparrow.visitor.Interpreter.visit(Interpreter.java:382)
< 	at sparrow.Call.accept(Call.java:19)
< 	at sparrow.visitor.Interpreter.stepUntilReturn(Interpreter.java:34)
< 	at sparrow.visitor.Interpreter.visit(Interpreter.java:97)
< 	at sparrow.Program.accept(Program.java:15)
< 	at Main.main(Main.java:35)
---
> 12
> 15
> 27
