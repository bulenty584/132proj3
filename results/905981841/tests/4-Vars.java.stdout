func Main()
  y0 = 0
  v0 = 4
  v1 = alloc(v0)
  v2 = 8
  vmt_A = alloc(v2)
  v3 = @A_helper
  [vmt_A + 0] = v3
  v4 = @A_run
  [vmt_A + 4] = v4
  [v1 + 0] = vmt_A
  if0 v1 goto v1Error_0
  goto v1End_1
  v1Error_0:
    error("null pointer")
  v1End_1:
    v5 = [v1 + 0]
    v6 = [v5 + 4]
    v7 = call v6(v1)
    print(v7)
    return y0
    func A_run(this)
      if0 this goto thisError_2
      goto thisEnd_3
      thisError_2:
        error("null pointer")
      thisEnd_3:
        v8 = 12
        v9 = [this + 0]
        v10 = [v9 + 0]
        v11 = call v10(this v8)
        a = v11
        if0 this goto thisError_4
        goto thisEnd_5
        thisError_4:
          error("null pointer")
        thisEnd_5:
          v12 = 15
          v13 = [this + 0]
          v14 = [v13 + 0]
          v15 = call v14(this v12)
          b = v15
          v16 = a + b
          return v16
        func A_helper(this param)
          x = param
          v17 = 1
          v18 = param + v17
          param = v18
          print(x)
          return x
